app/src/main/java/de/s3xy/architecturesample/login/ui/LoginActivity.java

```diff
@@ -0,0 +1,119 @@
+package de.s3xy.architecturesample.login.ui;
+
+import android.annotation.TargetApi;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.view.View;
+import android.webkit.WebResourceRequest;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.Unbinder;
+import de.s3xy.architecturesample.AwesomeApplication;
+import de.s3xy.architecturesample.R;
+import de.s3xy.architecturesample.base.BaseActivity;
+import de.s3xy.architecturesample.github.GithubAuthHelper;
+import de.s3xy.architecturesample.login.presenter.LoginPresenter;
+import de.s3xy.architecturesample.search.ui.SearchRepositoriesActivity;
+
+/**
+ * Created by Vlad Fedorenko <vfedo92@gmail.com>
+ * 08.25.2016
+ */
+public class LoginActivity extends BaseActivity implements LoginView {
+
+    @Inject
+    LoginPresenter mLoginPresenter;
+
+    @BindView(R.id.web_view)
+    WebView mWebView;
+
+    private Unbinder mUnbinder;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        ((AwesomeApplication) getApplication()).getApplicationComponent().inject(this);
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        mUnbinder = ButterKnife.bind(this);
+        mLoginPresenter.attachView(this);
+    }
+
+    protected Unbinder getUnbinder() {
+        return mUnbinder;
+    }
+
+    protected LoginPresenter getPresenter() {
+        return mLoginPresenter;
+    }
+
+    public void onSignInClick(View view) {
+        mLoginPresenter.signIn();
+    }
+
+    public void onSkipClick(View view) {
+        mLoginPresenter.skipLogin();
+    }
+
+    @Override
+    public void showWebViewForLogin(String url) {
```

Comment:
In my opinion a webview is always bad idea, because you leave the applicaiton. Github provides api endpoints for doing all authorization via api endpoints so the application can do the thinggs the webview does.
https://developer.github.com/v3/oauth_authorizations/